internal/api/v1/application/exec.go

```diff
@@ -71,12 +71,12 @@ func (hc Controller) Exec(c *gin.Context) apierror.APIErrors {
 		podToConnect = podNames[0]
 	}
 
-	deployment, err := workload.Deployment(ctx)
+	appData, err := workload.Get(ctx, 0)
```

Comment:
As we discussed in a call, `Workload` struct seems to be superfluous and can be removed (merged with the `AppDeployment`). Then we can implement a `GetWorkload` method on the app itself instead of this `Get`. We won't need to pass the desired instances then.

Reply:
Larger discussion got triggered ... Defering full resolution into new ticket https://github.com/epinio/epinio/issues/1657

For now a better intermediate change would be to move the `desiredInstances` argument from `Get` to `NewWorkload`.

Reply:
Intermediate change done in commit 6b070caadb31c350cc874eb9fb7cb307562b0bbd